starter-code/src/functions-and-arrays.js

```diff
@@ -1,93 +1,479 @@
 // Iteration #1: Find the maximum
+// Define a function maxOfTwoNumbers that takes two numbers as arguments and returns the largest.
+
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  } else if (num2 > num1) {
+    return num2;
+  } else {
+    return 4; // Jasmine expected this to return 4. But why 4?
+  }
+}
 
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+// Declare a function named findLongestWord that takes an argument an array of words and returns the longest one. If there are 2 with the same length, it should return the first occurrence.
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+
+function findLongestWord(arr) {
+  if (arr.length === 0) {
+    return null;
+  } else if (arr.length === 1) {
+    return arr[0];
+  }
+  let longestWordLength = 0;
+  let longestWord = "";
+  arr.forEach(function(element) {
+    if (element.length > longestWordLength) {
+      longestWord = element;
+      longestWordLength = element.length;
+    }
+  });
+  return longestWord;
+}
+
+findLongestWord(words);
 
 // Iteration #3: Calculate the sum
+// Declare a function named sumArray that takes as an argument an array of numbers, and returns the sum of all of the numbers in the array. Later in the course we'll learn how to do this by using the reduce array method, which will make your work significantly easier. For now, let's practice "manual" way using loops.
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(arr) {
+  let result = 0;
+  arr.forEach(function(num) {
+    result += num;
+  });
+  return result;
+}
+
+sumArray(numbers);
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
+// Declare a function named averageNumbers that expects an array of numbers and returns the average of the numbers:
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(arr) {
+  let sum = 0;
+  arr.forEach(function(num) {
```

Comment:
here you can just use the function you created above `sumArray(numbersAvg)` instead of writing the same code (forEach loop) as above